js/offer.js

```diff
@@ -0,0 +1,80 @@
+import {getHouseRentalCount} from './random-object.js';
+const cardTemplate = document.querySelector('#card').content.querySelector('.popup');
+const templateFRAGMENT = document.createDocumentFragment();
+const announcementMap = document.querySelector('.map__canvas');
+
+const mapTypeHousing = {
+  flat: 'Квартира',
+  bungalow: 'Бунгало',
+  house: 'Дом',
+  palace: 'Дворец',
+  hotel: 'Отель'
+};
+
+
+const getRandomFeatures = (array, list) => {
+  list.forEach((featureListItem) => {
+    const isNeedFeatures = array.some((someComfort) => featureListItem.classList.contains(`popup__feature--${someComfort}`));
+    if (!isNeedFeatures) {
+      featureListItem.remove();}
+  });
+};
+
+
+const getRandomPhotos = (array, container, element) => {
+  if (!array) {
+    container.remove();
+  } else {
+    array.forEach((arrayItem) => {
+      const photoFragment = element.cloneNode(true);
+      photoFragment.src = arrayItem;
+      templateFRAGMENT.appendChild(photoFragment);
+    });
+  }
+  return templateFRAGMENT;
+};
+
+const checkDataMissing = (data, element) => {
+  if (data && !data.includes('undefined')) {
+    if (data.indexOf('.png') >= 0) {
+      element.src = data;
+    } else {
+      element.textContent = data;
+    }
+  } else {
+    element.remove();
+  }
+};
+
+const returnSimilarCard = (data) => {
```

Comment:
never use return keyword in function name. Also dont understand what is similar card. Rename function